app/Domains/Vendors/Actions/Queries/GetVendors.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domains\Vendors\Actions\Queries;
+
+use App\Domains\Vendors\Vendor;
+use Illuminate\Auth\Access\Response;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+use Lorisleiva\Actions\ActionRequest;
+use Lorisleiva\Actions\Concerns\AsAction;
+
+class GetVendors
+{
+    use AsAction;
+
+    public function authorize(ActionRequest $request): Response
+    {
+        $user = $request->user();
+        
+        if ($user->has_general_access) {
+            return Response::allow();
+        }
+
+        return Response::deny('You are unauthorized to perform this action');
+    }
+
+    public function handle(
+        ?int $per_page = 10, 
+        ?string $search_term = ''
+    ): Collection | LengthAwarePaginator {
+        $query = Vendor::query();
+
+        if ($search_term) {
+            $query
+                ->where('name', 'like', "%$search_term%")
+                ->orWhere('contacts', 'like', "%$search_term%");
+        }
+    
+        return $per_page === null ?
+            $query->get() :
+            $query->paginate($per_page);
+    }
+
+    public function rules(): array
+    {
+        return [
+            'search_term' => ['nullable', 'string'],
+            'per_page' => ['nullable', 'integer', 'min:1'],
+        ];
+    }
+
+    public function asController(ActionRequest $request)
+    {
+        $search_term = $request->input('search_term');
```

Comment:
Please read this [comment](https://github.com/Eveneer/evengoams-backend/pull/63/files#r1749168825)